calico-vpp-agent/routing/routing_server.go

```diff
@@ -199,6 +205,26 @@ func (s *Server) serveOne() error {
 	return nil
 }
 
+func (s *Server) getBGPConf(log *logrus.Entry, clientv3 calicov3cli.Interface) (*calicov3.BGPConfigurationSpec, error) {
+	defaultBGPConf, err := s.getDefaultBGPConfig(log, clientv3)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting default BGP configuration")
+	}
+	nodeSpecificConf, err := clientv3.BGPConfigurations().Get(context.Background(), "node."+config.NodeName, options.GetOptions{})
+	if err == nil {
+		defaultBGPConf.ListenPort = nodeSpecificConf.Spec.ListenPort
```

Comment:
We should verify that these are set in the node-specific config before overriding the default value.
Also it's a bit more idiomatic go to arrange the code like this:
```
if (err != nil) {
    ... error handling
}
... success case
```